python/dask_cudf/dask_cudf/io/parquet.py

```diff
@@ -141,7 +145,7 @@ def write_metadata(parts, fmd, fs, path, append=False, **kwargs):
                 else _meta[0]
             )
             with fs.open(metadata_path, "wb") as fil:
-                _meta.tofile(fil)
+                fil.write(_meta.tobytes())
```

Comment:
Why do we need to call `tobytes()` here? Is this zero copy?

Reply:
Based on the docs: https://numpy.org/doc/stable/reference/generated/numpy.ndarray.tofile.html
```
When fid is a file object, array contents are directly written to the file, 
bypassing the file object’s write method. As a result, tofile cannot be used with files
objects supporting compression (e.g., GzipFile) 
or file-like objects that do not support fileno() (e.g., BytesIO).
```

This was causing issues for some remote filesystem objects as not all implement `fileno()` (but all do have the `write` method).

Note: `_meta` is a `NumPy.ndarray`

Reply:
Okay, I believe `bytes` is an owning object so `tobytes()` can't be zero copy. Can we use a memoryview or bytearray or something similar that doesn't involve an unnecessary memory copy?